modules/host/announce.go

```diff
@@ -78,5 +83,8 @@ func (h *Host) AnnounceAddress(addr modules.NetAddress) error {
 	if h.closed {
 		return errHostClosed
 	}
+
+	h.mu.Lock()
+	defer h.mu.Unlock()
```

Comment:
probably okay to throw this to the top, other functions in the host take the form

```
lock
defer unlock
resourcelock
defer resource unlock
if closed {
}

rest of function
```